lib/web_ui/dev/test_runner.dart

```diff
@@ -180,14 +178,55 @@ class TestCommand extends Command<bool> with ArgUtils {
       });
     }
 
+    await _buildTargets();
+
     if (runAllTests) {
-      await _runAllTests();
+      await _runAllTestsForCurrentPlatform();
     } else {
-      await _runTargetTests(targetFiles);
+      await _runSpecificTests(targetFiles);
     }
     return true;
   }
 
+  /// Builds all test targets that will be run.
+  Future<void> _buildTargets() async {
+    final Stopwatch stopwatch = Stopwatch()..start();
+    List<FilePath> allTargets;
+    if (runAllTests) {
+      allTargets = environment.webUiTestDir
+        .listSync(recursive: true)
+        .whereType<io.File>()
+        .where((io.File f) => f.path.endsWith('_test.dart'))
+        .map<FilePath>((io.File f) => FilePath.fromWebUi(path.relative(f.path, from: environment.webUiRootDir.path)))
+        .toList();
+    } else {
+      allTargets = targetFiles;
+    }
+
+    // Separate HTML targets from CanvasKit targets because the two use
+    // different dart2js options (and different build.*.yaml files).
+    final List<FilePath> htmlTargets = <FilePath>[];
+    final List<FilePath> canvasKitTargets = <FilePath>[];
+    final String canvasKitTestDirectory = path.join(environment.webUiTestDir.path, 'canvaskit');
+    for (FilePath target in allTargets) {
+      if (path.isWithin(canvasKitTestDirectory, target.absolute)) {
+        canvasKitTargets.add(target);
```

Comment:
question: in which platforms/browser are we expecting these targets to run? (1) now (2) in the best case future scenario

Reply:
All, except IE11.

Reply:
So these will when we re-enable felt test --browser=safari?

I think it will make my goal of reenabling those tests soon even harder than. Can you merge after I fix the Safari tests? Or you can add if MacOs skip type of a check right now.

Reply:
I checked. This test passes in Safari.